include/picongpu/plugins/transitionRadiation/TransitionRadiation.hpp

```diff
@@ -0,0 +1,661 @@
+
+        /** Amount of transition radiation values
+         *
+         * Calculates amount of different transition radiation values, which
+         * have to be computed.
+         *
+         * @return amount of transition radiation values to be calculated
+         */
+        static
+        unsigned int
+        elementsTransitionRadiation( )
+        {
+            return transitionRadiation::frequencies::nOmega * transitionRadiation::parameters::nObserver; // storage for amplitude results on GPU
+        }
+
+        /** Combine transition radiation data from each CPU and store result on master.
+         *
+         * @remark copyRadiationDeviceToHost( ) should be called before.
+         */
+        void
+        collectRadiationOnMaster( )
+        {
+            reduce(
+                nvidia::functors::Add( ),
+                tmpITR,
+                incTransRad->getHostBuffer( ).getBasePointer( ),
+                elementsTransitionRadiation( ),
+                mpi::reduceMethods::Reduce( )
+            );
+            reduce(
+                nvidia::functors::Add( ),
+                tmpCTRpara,
+                cohTransRadPara->getHostBuffer( ).getBasePointer( ),
+                elementsTransitionRadiation( ),
+                mpi::reduceMethods::Reduce( )
+            );
+            reduce(
+                nvidia::functors::Add( ),
+                tmpCTRperp,
+                cohTransRadPerp->getHostBuffer( ).getBasePointer( ),
+                elementsTransitionRadiation( ),
+                mpi::reduceMethods::Reduce( )
+            );
+            reduce(
+                nvidia::functors::Add( ),
+                tmpNum,
+                numParticles->getHostBuffer( ).getBasePointer( ),
+                elementsTransitionRadiation( ),
+                mpi::reduceMethods::Reduce( )
+            );
+        }
+
+        //! Write transition radiation data to file.
+        void
+        writeTransRadToText( )
+        {
+            // only the master rank writes data
+            if (isMaster)
+            {
+                // get time step as string
+                std::stringstream o_step;
+                o_step << currentStep;
+
+                // write totalRad data to txt
+                writeFile(theTransRad, folderTransRad + "/" + filenamePrefix + "_" + o_step.str( ) + ".dat");
+            }
+        }
+
+
+        //! perform all operations to get data from GPU to master
+        void
+        collectDataGPUToMaster( )
+        {
+            // collect data GPU -> CPU -> Master
+            copyRadiationDeviceToHost( );
+            collectRadiationOnMaster( );
+            sumTransitionRadiation( theTransRad, tmpITR, tmpCTRpara, tmpCTRperp, tmpNum );
+        }
+
+        /** Final transition radiation calculation on CPU side
+         *
+         * Calculate transition radiation integrals. This can't happen on the GPU
+         * since the absolute square of a sum can't be moved within a sum.
```

Comment:
I do not understand what you mean by "moved". Could you explain it here in GitHub?